### PR TITLE
REN-52/54: Spring Cloud

### PR DESCRIPTION
**Pull request by beeduck**
Completed Spring Cloud module. Added feign service for intermodule communication and oauth2 security between module communication.

### CODE REVIEW

File: Rent/Rent-Oauth/src/main/java/com/rent/auth/service/user/UserService.java [hunk 1]
```diff
@@ -11,5 +12,7 @@
 public interface UserService {
     UserDetails createUser(UserDTO userDTO, Locale locale);
 
+    void createUserInfoFallback(NewUserDTO newUserDTO);
```
Comment (AsadA128) [inline Rent/Rent-Oauth/src/main/java/com/rent/auth/service/user/UserService.java]: remove fall back

File: Rent/Rent-Util/pom.xml [hunk 2]
```diff
@@ -11,5 +11,11 @@
 
     <artifactId>Rent-Util</artifactId>
 
+    <dependencies>
```
Comment (AsadA128) [inline Rent/Rent-Util/pom.xml:14]: get rid of this

**Review by AsadA128 — CHANGES_REQUESTED** [review]
FIX THIS SHITTTTT!!!!

File: Rent/Rent-Util/pom.xml (hunk 2 above)
Comment (beeduck) [reply Rent/Rent-Util/pom.xml:14]: Will be staying, needed for the global exception handler

**Review by AsadA128 — APPROVED** [review]

**Comment by AsadA128** [comment]
good job Kris :)

**Pull request merged**